File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/keycloak/credential/UserNameCredential.java
package cc.coopersoft.accesstoken.keycloak.credential;

import cc.coopersoft.accesstoken.api.Credential;
import lombok.Builder;
import lombok.NoArgsConstructor;

import java.util.HashMap;
import java.util.Map;

@NoArgsConstructor
public class UserNameCredential implements Credential {

  private static final String USER_NAME_PARAM_NAME = "username";

  private static final String PASSWORD_PARAM_NAME = "password";

  Map<String,String> context = new HashMap<>(2);

  @Builder
  public UserNameCredential(String username, String password) {
    setPassword(password);
    setUserName(username);
  }

  @Override
  public Map<String, String> getContext() {
    return context;
  }

  public void setUserName(String userName){
    context.put(USER_NAME_PARAM_NAME,userName);
  }

  public String getUserName(){
    return context.get(USER_NAME_PARAM_NAME);
  }

  public void setPassword(String password){
    context.put(PASSWORD_PARAM_NAME,password);
  }

  public String getPassword(){
    return context.get(PASSWORD_PARAM_NAME);
  }
}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/keycloak/credential/UserNameCredential.kt
package cc.coopersoft.accesstoken.keycloak.credential

import cc.coopersoft.accesstoken.api.Credential

class UserNameCredential (username: String, password: String) : Credential {
    override val context: MutableMap<String, String> = HashMap(2)

    init {
        setPassword(password)
        setUserName(username)
    }

    fun setUserName(userName: String) {
        context[USER_NAME_PARAM_NAME] = userName
    }

    val userName: String?
        get() = context[USER_NAME_PARAM_NAME]

    fun setPassword(password: String) {
        context[PASSWORD_PARAM_NAME] = password
    }

    val password: String?
        get() = context[PASSWORD_PARAM_NAME]

    companion object {
        private const val USER_NAME_PARAM_NAME = "username"
        private const val PASSWORD_PARAM_NAME = "password"
    }
}